lib/shared/s3-packages.js

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const semver = require('semver');
+const Bluebird = require('bluebird');
+const request = Bluebird.promisifyAll(require('request'));
+const xml = Bluebird.promisifyAll(require('xml2js'));
+const release = require('./release');
+
+/**
+ * @summary Etcher S3 bucket URLs
+ * @namespace BUCKET_URL
+ * @public
+ */
+exports.BUCKET_URL = {
+
+  /**
+   * @property {String} PRODUCTION
+   * @memberof BUCKET_URL
+   * @description
+   * Etcher production S3 bucket URL
+   */
+  PRODUCTION: 'https://resin-production-downloads.s3.amazonaws.com',
+
+  /**
+   * @property {String} SNAPSHOT
+   * @memberof BUCKET_URL
+   * @description
+   * Etcher snapshot S3 bucket URL
+   */
+  SNAPSHOT: 'https://resin-nightly-downloads.s3.amazonaws.com'
+
+};
+
+/**
+ * @summary Etcher S3 package name
+ * @constant
+ * @private
+ * @type {String}
+ */
+const S3_PACKAGE_NAME = 'etcher';
+
+/**
+ * @summary Number of packages per Etcher version
+ * @constant
+ * @private
+ * @type {Number}
+ */
+const NUMBER_OF_PACKAGES = 8;
+
+/**
+ * @summary Get the correct S3 bucket url from a release type
+ * @function
+ * @private
+ *
+ * @param {RELEASE_TYPE} releaseType - release type
+ * @returns {(String|Null)} S3 bucket url
```

Comment:
See the "nullable type" section of http://usejsdoc.org/tags-type.html. Turns out we can do `{?String}` to signify that the result is either String or null.